plugins/mmdblookup/mmdblookup.c

```diff
@@ -133,6 +138,48 @@ void close_mmdb(MMDB_s *mmdb) {
 	MMDB_close(mmdb);
 }
 
+const char *SEP = "!";
+
+int mmdb_get_value(
+	MMDB_entry_s *const start,
+	MMDB_entry_data_s *const entry_data,
+	char *field,
+	size_t field_len
+) {
+	// Count names in the field to construct a path, e.g.,
+	// "!continent!code" implies 2 levels of JSON object {"continent": {"code": "XX"}}.
+	size_t length = 0;
+	for (size_t i = 0; i < field_len; i++) {
+		if (field[i] == *SEP) {
+			length++;
+		}
+	}
+
+	if (length == SIZE_MAX) {
+		return MMDB_INVALID_METADATA_ERROR;
```

Comment:
It seems confusing to return MMDB errors from this code. I would expect errors returned to be ones defined by rsyslog.

Reply:
I was thinking about `mmdb_get_value()` function as a thin wrapper over `MMDB_aget_value`, so the error handling would be as follows. Is that ok?

```c
mmdb_err = mmdb_get_value(&result.entry, &entry_data, field);
if (MMDB_SUCCESS != mmdb_err) {
	dbgprintf("Got an error looking up the entry data - %s\n", MMDB_strerror(mmdb_err));
	ABORT_FINALIZE(RS_RET_OK);
}
```